indico/htdocs/js/indico/modules/timetable/timetable/Draw.js

```diff
@@ -649,6 +649,31 @@ function loadBalloonContent(self, api, editable) {
                     }
                 });
             });
+
+            var $picker = $content.find('.palette-picker');
+            $picker.palettepicker({
+                availableColors: $picker.data('palette'),
+                selectedColor: $picker.data('initial-color'),
+                onSelect: function(background, text) {
+                    $.ajax({
+                        url: $picker.data('href'),
+                        method: $picker.data('method'),
+                        data: JSON.stringify({'colors': {'text': text, 'background': background}}),
+                        dataType: 'json',
+                        contentType: 'application/json',
+                        error: handleAjaxError,
+                        complete: function() {
+                            self.setColors(text, background);
+                            IndicoUI.Dialogs.Util.progress()();
```

Comment:
That doesn't do what you want - the point of passing `complete: IndicoUI.Dialogs.Util.progress()` is that the first invocation happens before the request is sent and then the returned function to close the dialog is executed when the request finished. So in this case you need to do call it before the `$.ajax()` call and assign the returned function and then call that one in here.